rest/config.go

```diff
@@ -807,6 +807,19 @@ func (config *StartupConfig) Serve(addr string, handler http.Handler) error {
 		http2Enabled,
 		tlsMinVersion,
 	)
+	if err != nil {
+		return err
+	}
+
+	sc.addHTTPServer(server)
+
```

Comment:
If someone called serverContext.Close at this point, it would attempt to Close() a server that hadn't run server.Serve() yet.  Is that a no-op/safe sequence of operations?

Reply:
Close is safe to call on a server that has not yet started. The server errors on startup after Close has been called. It's also safe to double close.

```go
func TestHTTPServerEarlyClose(t *testing.T) {
	s := http.Server{}
	// close before listen/serve
	fmt.Println(s.Close())
	// double close
	fmt.Println(s.Close())
	// listen/serve after close
	fmt.Println(s.ListenAndServe())
	fmt.Println(s.Close())
}
```
```
=== RUN   TestHTTPServerEarlyClose
<nil>
<nil>
http: Server closed
<nil>
--- PASS: TestHTTPServerEarlyClose (0.00s)
```